rec: Boost context is no longer detected with boost >= 1.72.0

- Program: Recursor
 - Issue type: Bug report

### Short description
On Arch with boost 1.72.0:
```
checking for the Boost context library... no
checking for getcontext... yes
checking for makecontext... yes
checking for swapcontext... yes
checking what context library to use for MTasker... System V ucontexts
```
```
configure:14335: checking for the Boost context library
configure:14401: g++ -c -Wall -Wextra -Wshadow -Wno-unused-parameter -g -O2    -pthread conftest.cpp >&5
conftest.cpp:43:10: fatal error: boost/context/execution_context_v2.hpp: No such file or directory
   43 | #include <boost/context/execution_context_v2.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This happens because our m4 detection tries to use `execution_context_v2.hpp` when the boost version is greater than or equal to 1.61, but unfortunately this API was removed in 1.72.0 beta 1: https://github.com/boostorg/context/commit/0c754f90c53ddb8d041c2ccbc78da5cb484c5f1f#diff-7750a754757adf2b59d3dd0cad775071

--- a/m4/boost.m4
+++ b/m4/boost.m4
@@ -22,7 +22,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 m4_define([_BOOST_SERIAL], [m4_translit([
-# serial 31
+# serial 32
 ], [#
 ], [])])
 
@@ -666,6 +666,8 @@ LDFLAGS=$boost_filesystem_save_LDFLAGS
 # * The implementation details were moved to boost::context::detail in 1.61.0
 # * 1.61 also introduces execution_context_v2, which is the "lowest common
 #   denominator" for boost::context presence since then.
+# * boost::context::fiber was introduced in 1.69 and execution_context_v2 was
+#   removed in 1.72
 BOOST_DEFUN([Context],
 [boost_context_save_LIBS=$LIBS
  boost_context_save_LDFLAGS=$LDFLAGS
@@ -676,7 +678,31 @@ if test $boost_major_version -ge 157; then
   LDFLAGS="$LDFLAGS $BOOST_THREAD_LDFLAGS"
 fi
 
-if test $boost_major_version -ge 161; then
+if test $boost_major_version -ge 169; then
+
+BOOST_FIND_LIB([context], [$1],
+                [boost/context/fiber.hpp], [[
+namespace ctx=boost::context;
+int a;
+ctx::fiber source{[&a](ctx::fiber&& sink){
+    a=0;
+    int b=1;
+    for(;;){
+        sink=std::move(sink).resume();
+        int next=a+b;
+        a=b;
+        b=next;
+    }
+    return std::move(sink);
+}};
+for (int j=0;j<10;++j) {
+    source=std::move(source).resume();
+}
+return a == 34;
+]], [], [], [$2])
+
+elif test $boost_major_version -ge 161; then
+
 BOOST_FIND_LIB([context], [$1],
                 [boost/context/execution_context_v2.hpp], [[
 namespace ctx=boost::context;

